tilings/algorithms/factor.py

```diff
@@ -254,5 +262,6 @@ def _unite_all(self) -> None:
         """
         Unite all the cells that share an obstruction or a requirement list.
         """
+        self._unite_assumptions()
```

Comment:
Not sure how the Interleaving and MonotoneInterleaving factor handle assumption. Maybe we should just raise an error if the tiling as assumptions in those cases?

Reply:
They should behave the same, although when using it, it could be beneficial for example, to have the splitting strategy use the relevant factor algorithm.